src/eth/eip_1559_transaction.rs

```diff
@@ -139,9 +139,25 @@ impl EthEip1559TxArgs {
         from: Address,
         eth_chain_id: EthChainId,
     ) -> anyhow::Result<SignedMessage> {
-        ensure!(self.chain_id != eth_chain_id, "Invalid chain id");
```

Comment:
@elmattic What was the purpose of this check here (why are we using `!=`) ?.

Reply:
Seems to be a type of sanity check at first glance, but the `!=` is puzzling, indeed. @sudo-shashank can you elaborate on that?

Reply:
I believe the equivalent code in Lotus is:
```
if tx.ChainID != buildconstants.Eip155ChainId {
    return nil, fmt.Errorf("invalid chain id: %d", tx.ChainID)
}
```

Reply:
Then I guess it was mistake because `ensure!` and `if` condition is not the same thing. No worries I have updated it will push the changes.

Reply:
Yeah, it looks like a mistake on our side to me.

Reply:
Done.